From: Chris Leech <cleech@redhat.com>
Subject: iscsiuio: allow ARP for non-matching src and dst addresses

Bugzilla: ZZZ
Upstream Status:
Build Info: XXX
Tested:

commit e43d687dc23e66c609491f9bfa8d8f29be7ef72d
Author: Nilesh Javali <nilesh.javali@cavium.com>
Date:   Thu Feb 22 07:25:58 2018 -0500

    iscsiuio: allow ARP for non-matching src and dst addresses
    
    For source and destination IP addresses in different
    networks, continue with the ARP retries and further login process
    instead of assuming abrupt failure. iSCSI offload adapters may not rely on
    netmask information for successful iSCSI target login.
    
    Signed-off-by: Nilesh Javali <nilesh.javali@cavium.com>
---
 iscsiuio/src/unix/libs/cnic.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/iscsiuio/src/unix/libs/cnic.c b/iscsiuio/src/unix/libs/cnic.c
index a009f25f0814..32166edf243f 100644
--- a/iscsiuio/src/unix/libs/cnic.c
+++ b/iscsiuio/src/unix/libs/cnic.c
@@ -362,9 +362,8 @@ int cnic_handle_ipv4_iscsi_path_req(nic_t *nic, int fd,
 			       &nic_iface->ustack.default_route_addr,
 			       sizeof(dst_addr));
 		} else {
-			arp_retry = MAX_ARP_RETRY;
-			LOG_DEBUG(PFX "%s: no default", nic->log_name);
-			goto done;
+			LOG_DEBUG(PFX "%s: no default route address",
+				  nic->log_name);
 		}
 	}
 	arp_retry = 0;
